store/secretsmgr_test.go

```diff
@@ -0,0 +1,496 @@
+		DeleteSecretCallCount:    0,
+		DeleteSecretLastParams:   secretsmanager.DeleteSecretInput{},
+		DeleteSecretOutput:       secretsmanager.DeleteSecretOutput{},
+	}
+}
+
+func (m *SecretsManagerClientMock) GetSecretValue(ctx context.Context, params *secretsmanager.GetSecretValueInput, optFunc ...func(options *secretsmanager.Options)) (*secretsmanager.GetSecretValueOutput, error) {
+	m.GetSecretValueCallCount++
+	m.GetSecretValueLastParams = *params //we intentionally _copy_ the params arg here in case it gets modified elsewhere
+	return m.GetSecretValueOutputStub(params.SecretId)
+}
+
+func (m *SecretsManagerClientMock) CreateSecret(ctx context.Context, params *secretsmanager.CreateSecretInput, optFunc ...func(*secretsmanager.Options)) (*secretsmanager.CreateSecretOutput, error) {
+	m.CreateSecretCallCount++
+	m.CreateSecretLastParams = *params //we intentionally _copy_ the params arg here in case it gets modified elsewhere
+
+	if m.CreateSecretError != nil {
+		return nil, m.CreateSecretError
+	} else {
+		rtnContent := m.CreateSecretOutput //again we intentionally copy the output here so as not to return a pointer to our own copy which can then be mutated
+		return &rtnContent, nil
+	}
+}
+
+func (m *SecretsManagerClientMock) PutSecretValue(ctx context.Context, params *secretsmanager.PutSecretValueInput, optFunc ...func(*secretsmanager.Options)) (*secretsmanager.PutSecretValueOutput, error) {
+	m.PutSecretValueCallCount++
+	m.PutSecretValueLastParams = *params //we intentionally _copy_ the params arg here in case it gets modified elsewhere
+	rtnContent := m.PutSecretValueOutput //again we intentionally copy the output here so as not to return a pointer to our own copy which can then be mutated
+
+	return &rtnContent, nil
+}
+
+func (m *SecretsManagerClientMock) DeleteSecret(ctx context.Context, params *secretsmanager.DeleteSecretInput, optFunc ...func(*secretsmanager.Options)) (*secretsmanager.DeleteSecretOutput, error) {
+	m.DeleteSecretCallCount++
+	m.DeleteSecretLastParams = *params //we intentionally _copy_ the params arg here in case it gets modified elsewhere
+	rtnContent := m.DeleteSecretOutput //again we intentionally copy the output here so as not to return a pointer to our own copy which can then be mutated
+
+	return &rtnContent, nil
+}
+
+func (m *SecretsManagerClientMock) ListSecrets(ctx context.Context, params *secretsmanager.ListSecretsInput, optFunc ...func(*secretsmanager.Options)) (*secretsmanager.ListSecretsOutput, error) {
+
+	m.ListSecretsLastParams = *params //we intentionally _copy_ the params arg here in case it gets modified elsewhere
+	idx := m.ListSecretsCallCount
+	if idx > len(m.ListSecretsOutputs) {
+		idx = len(m.ListSecretsOutputs) - 1
+	}
+	rtnContent := m.ListSecretsOutputs[idx] //again we intentionally copy the output here so as not to return a pointer to our own copy which can then be mutated
+
+	m.ListSecretsCallCount++
+
+	return &rtnContent, nil
+}
+
+func TestSecretsManager_Get(t *testing.T) {
```

Comment:
❤️  the effort to test the SecretsManager store here.

Reply:
I want to be slightly contrary here. I'm not sure how much value these tests are adding? And the model is perhaps confusing here:

```
SecretsManagerClientMock // hand-rolled mocking structure for the tests
SecretsManagerClientIface // a narrowing of the secretsmanager.Client (AWS) interface
SecretsManager // the 'Store' implementation
secretsmanager.Client // AWS client def
```

`SecretsManagerClientIface`: is only used internally by `SecretsManager` so it's not clear what value it brings - to make explicit which parts of SM should be used yes, but the Store interface should be enough here to guide things (i.e. in practice you need to use whatever is necessary to satisfy Store so the `SecretsManagerClientIface` will just end up mirroring that).

`SecretsManagerClientMock`: really we are just testing out mock implementation here. I.e. it is testing itself! So I'd probably suggest we scrap these, especially as it is a lot of code. The truth is, It's pretty tricky to effectively test IO/client calls like these cases, where there is no data transformation. (On the other hand - the `store_test.go` tests you added are definitely very helpful.) We would probably get much more value by adding one or two integration tests here with actual SM.